lib/rubocop/cop/style/stabby_lambda_parentheses.rb

```diff
@@ -0,0 +1,108 @@
+# encoding: utf-8
+module RuboCop
+  module Cop
+    module Style
+      # Check for parentheses around stabby lambda arguments.
+      # There are two different styles. Defaults to `require_parentheses`.
+      #
+      # @example
+      #   # require_parentheses - bad
+      #   ->a,b,c { a + b + c }
+      #
+      #   # require_parentheses - good
+      #   ->(a,b,c) { a + b + c}
+      #
+      #   # require_no_parentheses - bad
+      #   ->(a,b,c) { a + b + c }
+      #
+      #   # require_parentheses - good
+      #   ->a,b,c { a + b + c}
+      class StabbyLambdaParentheses < Cop
+        MSG_REQUIRE = 'Wrap stabby lambda arguments with parentheses.'
+        MSG_NO_REQUIRE = 'Do not wrap stabby lambda arguments with parentheses.'
+
+        include ConfigurableEnforcedStyle
+
+        def on_send(node)
+          return unless lambda_with_args?(node)
+
+          if style == :require_parentheses
+            if !parentheses?(node)
+              missing_parentheses(node)
+            else
+              correct_style_detected
+            end
+          elsif parentheses?(node)
+            unwanted_parentheses(node)
+          else
+            correct_style_detected
+          end
+        end
+
+        def autocorrect(node)
+          if style == :require_parentheses
+            missing_parentheses_corrector(node)
+          elsif style == :require_no_parentheses
```

Comment:
It could be. I feel though its important to keep some indication between the style and the corrector. Would a case statement be better?